server/gokbg3/grails-app/domain/org/gokb/cred/Identifier.groovy

```diff
@@ -1,196 +1,184 @@
 package org.gokb.cred
 
-import gokbg3.MessageService
-import grails.util.Holders
-import groovy.transform.Synchronized
-import org.grails.web.json.JSONArray
-import org.grails.web.json.JSONObject
-
-import javax.persistence.Transient
-import java.util.regex.Pattern
 import groovy.util.logging.*
+import org.aspectj.weaver.ast.Or
 
 
 @Slf4j
-class Identifier extends KBComponent {
-
-  static def messageService
+class Identifier {
 
   IdentifierNamespace namespace
   String value
+  String uuid
 
-  @Override
-  Collection<String> getLogIncluded() {
-    ['value']
-  }
-
-  @Override
-  String getLogEntityId() {
-    "${this.class.name}:${id}"
-  }
-
-  public String getRestPath() {
-    return "/identifiers"
-  }
+  // Timestamps
+  Date dateCreated
+  Date lastUpdated
 
-  static jsonMapping = [
-    'ignore'      : [
-      'lastUpdatedBy',
-      'dateCreated',
-      'name',
-      'status',
-      'lastUpdated',
-      'description',
-      'source',
-      '_links'
-    ],
-    'defaultLinks': [
-      'namespace'
-    ]
+  static belongsTo = [
+          tipp: TitleInstancePackagePlatform,
+          org: Org,
+          platform: Platform,
+          pkg: Package
   ]
 
+
   static constraints = {
+    uuid(nullable: true, unique: false, blank: false, maxSize: 2048)
```

Comment:
UUIDs should never be nullable; nully entries should be remapped

Reply:
I have to run a separate action for this. Using SQL to set the UUID is not that trivial